qiita_pet/handlers/api_proxy/sample_template.py

```diff
@@ -0,0 +1,266 @@
+            if warns:
+                msg = '; '.join([convert_text_html(str(w.message))
+                                 for w in warns])
+                status = 'warning'
+
+    except Exception as e:
+        # Some error occurred while processing the sample template
+        # Show the error to the user so they can fix the template
+        status = 'error'
+        msg = str(e)
+    return {'status': status,
+            'message': msg,
+            'file': sample_template}
+
+
+@execute_as_transaction
+def sample_template_put_req(study_id, user_id, sample_template):
+    """Updates a sample template using the given file
+
+    Parameters
+    ----------
+    study_id : int
+        The current study object id
+    user_id : str
+        The current user object id
+    sample_template : str
+        filepath to use for updating
+
+    Returns
+    -------
+    dict
+        results dictonary in the format
+        {'status': status,
+         'message': msg,
+         'file': sample_template}
+
+    status can be success, warning, or error depending on result
+    message has the warnings or errors
+    file has the file name
+    """
+    access_error = check_access(study_id, user_id)
+    if access_error:
+        return access_error
+    # Define here the message and message level in case of success
+    status = "success"
+    # Get the uploads folder
+    _, base_fp = get_mountpoint("uploads")[0]
+    # Get the path of the sample template in the uploads folder
+    fp_rsp = join(base_fp, str(study_id), sample_template)
+
+    if not exists(fp_rsp):
+        # The file does not exist, fail nicely
+        return {'status': 'error',
+                'message': 'file does not exist',
+                'file': sample_template}
+
+    msg = ''
+    try:
+        with warnings.catch_warnings(record=True) as warns:
+            # deleting previous uploads and inserting new one
+            st = SampleTemplate(study_id)
+            df = load_template_to_dataframe(fp_rsp)
+            st.extend(df)
+            st.update(df)
+            remove(fp_rsp)
+
+            # join all the warning messages into one. Note that this info
+            # will be ignored if an exception is raised
+            if warns:
+                msg = '\n'.join(set(str(w.message) for w in warns))
+                status = 'warning'
+
+    except Exception as e:
+            status = 'error'
+            msg = str(e)
+    return {'status': status,
+            'message': msg,
+            'file': sample_template}
+
+
+@execute_as_transaction
+def sample_template_delete_req(study_id, user_id):
+    """Deletes the sample template attached to the study
+
+    Parameters
+    ----------
+    study_id : int
+        The current study object id
+    user_id : str
+        The current user object id
+
+    Returns
+    -------
+    dict
+        results dictonary in the format
+        {'status': status,
+         'message': msg}
+
```

Comment:
Wow. That's confusing :S

I'll eat all my words

On Wed, Dec 30, 2015 at 1:27 PM, Joshua Shorenstein <
notifications@github.com> wrote:

> In qiita_pet/handlers/api_proxy/sample_template.py
> <https://github.com/biocore/qiita/pull/1582#discussion_r48629045>:
>
> > +    """Deletes the sample template attached to the study
> > +
> > +    Parameters
> > +    ----------
> > +    study_id : int
> > +        The current study object id
> > +    user_id : str
> > +        The current user object id
> > +
> > +    Returns
> > +    -------
> > +    dict
> > +        results dictonary in the format
> > +        {'status': status,
> > +         'message': msg}
> > +
>
> Please look at what the code is calling, as it is not that function. It is this
> one
> <https://github.com/squirrelo/qiita/blob/artifact-study-api/qiita_pet/handlers/api_proxy/util.py#L6>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1582/files#r48629045>.
>